src/components/compound/DlCharts/charts/DlConfusionMatrix/DlConfusionMatrix.vue

```diff
@@ -0,0 +1,520 @@
+                ref="colorSpectrum"
+                class="color-spectrum"
+            >
+                <div class="color-spectrum__gradient" />
+                <div class="color-spectrum__gradation">
+                    <div
+                        v-for="value in gradationValues"
+                        :key="value"
+                        class="color-spectrum__gradation--element"
+                    >
+                        <span class="color-spectrum__gradation--element-line">-</span>
+                        {{ value !== gradationValues[0] ? value : null }}
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div
+            v-else
+            class="invalid"
+        >
+            The given props cannot create a valid matrix.
+        </div>
+        <div
+            v-if="tooltipState?.visible"
+            :style="tooltipStyles"
+            class="tooltip"
+        >
+            <span class="tooltip__x">{{ tooltipState.xLabel }}</span>
+            <span class="tooltip__y">{{ tooltipState.yLabel }}</span>
+            <span>
+                <span
+                    v-if="tooltipState.value"
+                    class="tooltip__color"
+                    :style="`background-color: ${getCellBackground(
+                        tooltipState.value
+                    )}`"
+                />
+                {{ normalized ? 'Normalized' : 'Unnormalized' }}
+                {{ tooltipState.value }}
+            </span>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue-demi'
+import DlBrush from '../../components/DlBrush.vue'
+import DlTooltip from '../../../../essential/DlTooltip/DlTooltip.vue'
+import { MatrixCell, Label, BrushState } from './types'
+import { hexToRgbA } from '../../../../../utils/colors'
+import { colorNames } from '../../../../../utils/css-color-names'
+import { useThemeVariables } from '../../../../../hooks/use-theme'
+import {
+    getGradationValues,
+    normalizeMatrix,
+    validateMatrix,
+    setZoom,
+    getCellWidth
+} from './utils'
+import { debounce } from 'lodash'
+export default defineComponent({
+    components: {
+        DlBrush,
+        DlTooltip
+    },
+    props: {
+        labels: {
+            required: true,
+            type: Array as PropType<string[] | Label[]>,
+            default: (): [] => []
+        },
+        matrix: {
+            required: true,
+            type: Array as PropType<number[][]>,
+            default: (): [] => []
+        },
+        normalized: {
+            type: Boolean,
+            default: true
+        },
+        color: {
+            type: String,
+            default: '--dl-color-secondary'
+        },
+        bottomLabel: {
+            type: String,
+            default: 'Predicted Label'
+        },
+        leftLabel: {
+            type: String,
+            default: 'True Label'
+        },
+        getLink: {
```

Comment:
why do we need a function here ?
shouldnt the link be part of the cell data ?

Create a new type of matrix input where its not just a number[][] but a {value: number, link: string}[][] 
one of those fields is the link


since we are exporting the matrixCell type why not allow people to give you it straight up ? 
that should also fix the function  as a prop issue.

Create a `type DlMatrix = number[][] | { value: number, link?: string }[][] | DlMatrixCell[]`
use that as the matrix type in this case

Reply:
> why do we need a function here ? shouldnt the link be part of the cell data ?
> 
> Create a new type of matrix input where its not just a number[][] but a {value: number, link: string}[][] one of those fields is the link
> 
> since we are exporting the matrixCell type why not allow people to give you it straight up ? that should also fix the function as a prop issue.
> 
> Create a `type DlMatrix = number[][] | { value: number, link?: string }[][] | DlMatrixCell[]` use that as the matrix type in this case

Would that mean each cell would hold an individual link? I passed it a function in order to build the link logic from outside using the cell labels and Id. You can build a function outside that accepts an id and return any kind of link for the cell containing that id:
```
 function getLink (cell: MatrixCell) => {
            return `www.confusion-matrix.com/cell-${cell.xLabel}-${cell.yLabel}-${cell.value}`
        }
```

Reply:
i understand but today we have a different pattern for passing a link to a confusion matrix table

We pass either ```number[][]```
or we pass an array of objects in the form of ```{ value: number, link?: string }[][]```

we do not need to pass a function to generate or get the link at all.

the link should be passed by the data.